lib/sentry/opentelemetry/span_record.ex

```diff
@@ -0,0 +1,72 @@
+if Code.ensure_loaded?(OpenTelemetry) do
+  defmodule Sentry.OpenTelemetry.SpanRecord do
+    require Record
+    require OpenTelemetry
+
+    @fields Record.extract(:span, from_lib: "opentelemetry/include/otel_span.hrl")
+    Record.defrecordp(:span, @fields)
+
+    defstruct @fields ++ [:origin]
+
+    def new(otel_span) do
```

Comment:
Can we add specs to all public functions like we do in the rest of the codebase?

Reply:
Also, you can add an additional check here

```suggestion
    def new(span() = otel_span) do
```